app/models/exam.rb

```diff
@@ -129,4 +129,21 @@ def self.remove_previous_version(eid, guide_id)
       exams.destroy_all
     end
   end
+
+  def submission_context_for(assignment)
+    (super if has_attempts_left?(assignment)) || assignment
+  end
+
+  def has_attempts_left?(assignment)
+    assignment.attempts_left > 0
```

Comment:
We are not dealing with nils here

Reply:
This method is only called when the exercise is capped, meaning attempts_left won't be nil.
Nevertheless, i agree we should deal with nils to avoid future problems.